standards/chronologic-piper.md

```diff
@@ -0,0 +1,124 @@
+# Specification
+
+## Summary
+A standardized interface for interacting with subscriptions.
+
+##  Abstract
+This standard defines a contract interface for interacting with subscription data. The scope of this proposal is end-user / wallet interactions. Creation of the subscriptions is out-of-scope of this ERC.
+
+## Our Thoughts
+From the subscriber's side:
+
+* I want strong guarantees on when I can cancel my subscription.
+* I will normally want payments to happen automatically without any action on my part.
+* In some cases it might be valuable to require an approval process.
+* For dynamically priced subscriptions I want to be able to set limits (require authorization if subscription is more than X).
+
+
+From the providers's side:
+
+* I need the ability to charge a fixed fee per subscription time unit (netflix, pandora, etc).
+* I need the ability to charge a dynamic fee per subscription time unit (aws, twilio, etc).
+* I need to be able to create reasonably accurate forecasts for upcoming subscriptions: Programatic checks that subscription accounts have available balance and that subscription is active.
+
+## Motivation
+Standard that covers all usecases of subscription based model with minimal interface suitable for wallets.
+
+### Interface
+
+#### cancel
+
+Immediately cancels the subscription. Implementations should ensure that any unpaid subscription payments are paid to the provider as part of cancellation to ensure providers are able to let subscriptions fees fill up for arbitrary lengths of time, allowing them to reduce overhead from transaction costs.
```

Comment:
I don't think we can ensure any payment executions since there are many different scenarios when the payment execution is not possible (eg. subscriber has insufficient funds).

Reply:
We can ensure unclaimed funds are paid if there are sufficient funds though. If there are not sufficient funds, perhaps there is a debt pattern that we can employ here. It may be out of scope for the standard but could be outlined as a suggested option.

Reply:
I would suggest leaving this for the provider / processor to handle. For instance, they could postpone a payment N-times until finally cancelling the subscription.
Subscriptions in fiat for example also cannot ensure the actual payment.

Reply:
Agree that's a little bit tricky. I was thinking to combine this with `isFunded` and/or `availableFunds` function described above.

So the scenario from the provider perspective is:
* provider monitors isFunded/availableFunds - this tells current funding status
* based on that information we can make a decision wether continoue the service (as @captnseagraves mentioned, provide the debt) or cancel